### DISCUSSION TITLE
Multiple rooms Jamulus Mac OS

### DISCUSSION THREAD

**Discussion by TheRico12** [jamulussoftware/jamulus#1473] [Technical Support & Help] [answered]
Hi, I'm trying to setup multiple rooms on my Jamulus server that I'm running on my Mac. I found a couple of guides for Linux cloud based solution, but nothing for Mac OS.

Thanks in advance

**Comment by gilgongo** [comment, marked as answer]
Hi - you can set up multiple servers on the command line with the `--port` option. So to have one on port 22125 for example:

`/Applications/Jamulus.app/Contents/MacOS/Jamulus -s --port 22125 -n -e rock.jamulus.io:22424 -o "yourServerName1;yourCity;[country ID]"`

You will need to tell people to use that port number when they connect though (eg xxx.xxx.xxx.xxx:22125) and need port forwards for both servers of course. See [this page](https://jamulus.io/wiki/Command-Line-Options) for the other options. 

I think, but I'm not sure, that you may get port-forwarding or NAT issues with some routers when you do this, but give it a go.

There is also the method of creating a custom directory server too, but that's only worth the effort if you want loads of rooms (and have loads of bandwidth for them).

**Reply by TheRico12** [reply]
@gilgongo

Hi! Thank you so much for your quick and clear reply.

For what I’ve understood (sorry but I’m not so friendly with the terminal) with your suggestion I will create a public server, is it right? Is there a way to create a private one?

The second option you were talking about refers to creating  a Custom Central server?
I’m trying to create a server for a music school where I’m teaching at, and it would be really useful having multiple rooms. 

Thanks again!

TheRico

**Reply by TheRico12** [reply]
@gilgongo 

i just tried just with 

/Applications/Jamulus.app/Contents/MacOS/Jamulus -s --port 22125

and it worked perfectly! It opened a new private instance of the Jamulus Server.
But I think that in my case scenario, it would be better to have this rooms attached to a Custom Central Server.

**Reply by gilgongo** [reply]
Ah yes, sorry, the example I gave was for a public server by mistake as I clipped it from the docs.

If you want to set up a custom directory (BTW "directory server" is the new preferred term for this - we'll be updating that soon!) then first be aware that you need to tell all your users to input that in their "customer central server address" field in their settings. 

Then set up a directory/central server as per the instruction for that here https://jamulus.io/wiki/Choosing-a-Server-Type

You may need to experiment a bit with router settings. I would open all UDP ports to the central server, for example. Note also that the  `--serverpublicip` option will probably be necessary for the servers you run but it's a new feature and I've not used it myself.

**Reply by TheRico12** [reply]
@gilgongo 

Thank you so much, I'll give it a try!

**Reply by TheRico12** [reply]
@gilgongo Thank you so much! Everything is working, other than that when I launch a new instance and specify --serverinfo [name];[city];[locale value], they don't remain stored in the GUI, I have to insert them manually. Where am I going wrong?

Thank you again for your precious help!

**Reply by gilgongo** [reply]
I think you may need to specify `--inifile` (see https://jamulus.io/wiki/Command-Line-Options) when you start each server, so that they can store their settings in a separate file. It doesn't matter where you store them BTW.

**Reply by TheRico12** [reply]
@gilgongo sorry, I’m a bit dam...I think I have understood, but could you please give me an example of it? So I could type —inifile and then the destination where I want to save the file?

**Reply by gilgongo** [reply]
Yes, so if you create a directory somewhere and set the path to it like this:

`/Applications/Jamulus.app/Contents/MacOS/Jamulus -s --inifile /some/path/here/server-1.ini`

Then when the server starts, all its config will be stored in that file.

**Reply by TheRico12** [reply]
@gilgongo thank you so much!!
